assignment3.rb

```diff
@@ -0,0 +1,143 @@
+require 'csv'
+
+class Employee
+  attr_reader :name, :age, :department, :division, :designation
+
+  def initialize(name, age, department, division, designation)
+    @name = name
+    @age = age
+    @department = department
+    @division = division
+    @designation = designation
+  end
+
+  def to_csv_row
+    [name, age, department, division, designation]
+  end
+end
+
+# Subclasses for different divisions
+class DevelopmentEmployee < Employee; end
+class QAEmployee < Employee; end
+class DevopsEmployee < Employee; end
+
+class EmployeeFactory
+  def create_employee(name, age, department, division, designation)
+    case division
+    when 'Development'
+      DevelopmentEmployee.new(name, age, department, division, designation)
```

Comment:
if you are already checking with division, no need to pass division to the constructor while initializing.

Reply:
removed devision